docs/designs/conceptual-versioning.md

```diff
@@ -0,0 +1,571 @@
+
+    A better user experience sample when using the new config:
+
+    If there is a file `articles/folder1/a.md` with monikerRange `'netcore-1.0'`, and file `articles/folder2/b.md` with monikerRange `'netcore-2.0'`, and you have to add a file `c.md` with monikerRange `'netcore-1.0 || netcore-2.0'`.
+
+    In docfx v2, since you cannot have overlap monikerRange in config, you have to copy `c.md` to `articles/folder1/` and `articles/folder2/`, which will bring you more maintenance cost for `c.md`.
+
+    But in docfx v3, you just need to maintenance one `c.md`
+
+#### 2.2 Markdown file
+
+##### 2.2.1 YAML header
+
+For conceptual markdown file, file level monikerRange setting is supported by using YAML header.
+
+```markdown
+---
+monikerRange: >=netcore-1.0
+---
+```
+
+> [!NOTE]
+> The final file level moniker range is the intersection of moniker range from `config file` and moniker range from `YAML header`, if the intersection is empty, an error throws.
+
+##### 2.2.2 Moniker zone
+
+Inside the file, with help of [versioned zone syntax](https://review.docs.microsoft.com/en-us/new-hope/resources/conceptual-versioning?branch=master#configuring-markdown-files-with-versioned-zones), the file has ability to wrap versioned content into different zones.
+
+```markdown
+---
+monikerRange: >=netcore-1.0
+---
+
+# Moniker Zone sample
+
+content in `>=netcore-1.0`
+
+::: moniker range="netcore-1.0"
+content just in `netcore-1.0`
+::: moniker-end
+
+::: moniker range=">netcore-1.0"
+content just in `>netcore-1.0`
+::: moniker-end
+```
+
+> [!NOTE]
+> The final zone moniker range is the intersection of moniker range of this file and moniker range of this zone. If the intersection is empty, an error throws.
+
+##### 2.2.3 Link/Xref
+
+In phase 1, content writer is allowed to reference another file with specific version by query string `?view={moniker}`, and the query string will be preserved to the resolve result, but we will not do the version existed check, and DHS will handle the fallback logic.
+
+There are several scenarios:
+
+|Reference type                                                                 |Resolve result             |Validation                 |
+|-------------------------------------------------------------------------------|---------------------------|---------------------------|
+|relative link without verison query - `[B](b.md)` |href without query - `<a href="b">` |file existed check |
+|relative link with verison query - `[B](b.md?view=netcore-1.0)` |href with original query - `<a href="b?view=netcore-1.0">` |file existed check|
+|absolute link/external link without verison query - `[B](/b.md)` |href without query - `<a href="/b">` |no validation |
+|absolute link/external link with verison query - `[B](/b.md?view=netcore-1.0)` |href with original query - `<a href="/b?view=netcore-1.0">` |no validation |
+|internal/external xref without verison query - `<xref: b>` |href without query - `<a href="b">` |uid existed check |
```

Comment:
need to say that we don't validate whether the uid exists in that particular moniker